src/codegen_lambda.cpp

```
@@ -501,4 +502,232 @@ std::string CodeGen::reverseResolveTypeName(llvm::Type *ty) {
+    uniform_closure_dtor_ = dtorFn;
+    return dtorFn;
+}
+
+llvm::Value *CodeGen::wrapAsUniformClosure(llvm::Value *val, const FnTypeInfo &info) {
+    // Already a uniform closure — pass through
+    if (info.isUniformClosure)
+        return val;
+
+    auto *ucTy = getUniformClosureTy();
+
+    llvm::Function *thunk = nullptr;
```

Clang (14.0.6)

[warning] variable 'thunk' is not initialized

(cppcoreguidelines-init-variables)